.github/workflows/smoke-test.yml

```
@@ -0,0 +1,43 @@
+name: smoke-test
+
+on:
+  push:
+    branches: ['main']
+
+jobs:
+  fetch-target:
+    name: Fetch Target From TUF Repo
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - 14.17.0
+          - 14.x
+          - 16.13.0
+          - 16.x
+          - 18.0.0
+          - 18.x
+        platform:
+          - os: ubuntu-latest
+            shell: bash
+          - os: macos-latest
+            shell: bash
+          - os: windows-latest
+            shell: cmd
+    runs-on: ${{ matrix.platform.os }}
+    defaults:
+      run:
+        shell: ${{ matrix.platform.shell }}
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
+      - name: Setup node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: npm
+      - name: Install dependencies
+        run: npm ci
+      - name: Run TUF client with Sigstore TUF repo
+        run: |
+          npx ts-node ./scripts/tuf_client_fetch_target.ts ./scripts/sigstore-tuf-repo-config.json
```

This is a small nitpick but the snake case is hurting my eyes.. 😄  js/ts files are usually camel cased:
```suggestion
          npx ts-node ./scripts/tufClientFetchTarget.ts ./scripts/sigstore-tuf-repo-config.json
```